### PR TITLE
Clarify network byte order conversions for integer / IP address conversions.

### PR DESCRIPTION
**Pull request by frewsxcv**
Opened primarily to address https://github.com/rust-lang/rust/issues/48819.

Also added a few other conversion docs/examples.

### CODE REVIEW

File: src/libstd/net/ip.rs [hunk 1]
```diff
@@ -769,7 +769,16 @@ impl FromInner<c::in_addr> for Ipv4Addr {
 
 #[stable(feature = "ip_u32", since = "1.1.0")]
 impl From<Ipv4Addr> for u32 {
-    /// It performs the conversion in network order (big-endian).
+    /// Convert an `Ipv4Addr` into a host byte order `u32`.
```
Comment (frewsxcv) [inline src/libstd/net/ip.rs]: admittedly i'm not sure what this terminology should be here w.r.t. the removal of 'network order' and addition of 'host byte order'. i just took it as a suggestion from [this thread](https://github.com/rust-lang/rust/issues/48819). might be nice to get input from the libs/docs team here @rust-lang/libs @rust-lang/docs

Comment (TimNN) [reply src/libstd/net/ip.rs]: To me the new wording seems good, and more easily understandable than the old version, although I'd be curious to hear what other people think.

File: src/libstd/net/ip.rs [hunk 2]
```diff
@@ -778,21 +787,48 @@ impl From<Ipv4Addr> for u32 {
 
 #[stable(feature = "ip_u32", since = "1.1.0")]
 impl From<u32> for Ipv4Addr {
-    /// It performs the conversion in network order (big-endian).
+    /// Convert an host byte order `u32` into an `Ipv4Addr`.
```
Comment (TimNN) [inline src/libstd/net/ip.rs]: nit: I think this should be "***a*** host byte [...]" (At least it should be consistent with the doc comment above).

Comment (frewsxcv) [reply src/libstd/net/ip.rs]: good catch! addressed in the latest force push

**Comment by clarcharr** [comment]
Why doesn't this include the `u128` impls as well?

File: src/libstd/net/ip.rs (hunk 1 above)
Comment (BurntSushi) [reply src/libstd/net/ip.rs]: I agree. The new wording looks good, and this is how I'd expect this function to behave.

**Comment by frewsxcv** [comment]
> Why doesn't this include the `u128` impls as well?

ran out of time when i made this branch. we can expand to the the u128 impls in a future pr

**Pull request merged**